packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -971,11 +971,16 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 /// A [Navigator] observer that notifies [RouteAware]s of changes to the
 /// state of their [Route].
 ///
+/// ## Sample code
+///
 /// To make a [StatefulWidget] aware of its current [Route] state, implement
 /// [RouteAware] in its [State] and subscribe it to the [RouteObserver]:
 ///
 /// ```dart
 /// class RouteAwareWidgetState extends State<RouteAwareWidget> with RouteAware {
+///
+///   final RouteObserver routeObserver = new RouteObserver();
```

Comment:
Done, but I think that made the example worse. Previously, the example had a clear focus: How do I implement RouteAware correctly in a widget. Now, the example is convoluted with other stuff (e.g. how to register the RouteObserver with the Navigator, something that is not specific to the RouteObserver and common to all NavigatorObservers).